### DISCUSSION TITLE
Add the possibility to change the column title - ColumnHelpers setTitle()

### DISCUSSION THREAD

**Discussion by spajz** [rappasoft/laravel-livewire-tables#978] [Feature Requests]
Hi, is there possibility to change column title after the creation of column?

**Comment by lrljoe** [comment]
Bit of context here, are you just referring to the label that is displayed in the header?

For example one of my columns looks like this:
```
            Column::make(__('contacttype.name'), 'name')
                ->sortable(),

```
The displayed title is taken from a localized value for the contacttype name.

If you're not into localized values then this would just display Contact Name"
```
            Column::make('Contact Name', 'name')
                ->sortable(),

```

What's your use case for changing it?

**Reply by spajz** [reply]
One of the reasons is add the flag (magnifying glass) if the column is set as searchable. I know you can change the name when creating a column, but this would work automatically by changing the columns array after initialization through Livewire hooks.
![image](https://user-images.githubusercontent.com/801824/209859701-4fb5ddaa-0cf3-42ef-98d4-98df85241dd1.png)

**Reply by lrljoe** [reply]
Two opttions:

Use setThAttributesin your configure option, then use CSS to add the icon.
```
public function configure(): void
{
  // Takes a callback that gives you the current column.
        $this->setThAttributes(function(Column $column) {
            if ($column->isSearchable()) {
              return [
                'class' => 'searchIcon',
              ];
            }
            return [];
          });
}
```

OR

You can publish the views and customise the th views, e.g. in components.table.th.blade.php add something like this:

```
@if($column->isSearchable())  
HourGlass Here
@endif
```
